core/src/test/java/org/apache/iceberg/rest/TestRESTCatalog.java

```diff
@@ -2506,6 +2506,29 @@ public void testNamespaceExistsViaHEADRequest() {
             any());
   }
 
+  @Test
+  public void renameTableNamespaceWithCatalogName() {
```

Comment:
please move this to `CatalogTests` so that we can test the same behavior with different catalog implementations

Reply:
Done,  only RestCatalog(maybe Hive also?) handles this case for now.